gen/_template/schema/stream.tmpl

```diff
@@ -1,17 +1,27 @@
 {{- /*gotype: github.com/ogen-go/ogen/gen/ir.Type*/ -}}
 {{- define "schema/stream" }}
 type {{ $.Name }} struct {
-	Data io.Reader
+	Data io.ReadCloser
 }
 
 // Read reads data from the Data reader.
 //
-// Kept to satisfy the io.Reader interface.
+// Kept to satisfy the io.ReadCloser interface.
 func (s {{ $.ReadOnlyReceiver }}) Read(p []byte) (n int, err error) {
 	if s.Data == nil {
 	    return 0, io.EOF
 	}
 	return s.Data.Read(p)
 }
 
+// Close closes the Data reader.
+//
+// Kept to satisfy the io.ReadCloser interface.
+func (s {{ $.ReadOnlyReceiver }}) Close() error {
```

Comment:
Also, there is a problem with multi-type responses.

For now, if spec defines multiple response types, `ogen` would generate an interface for the response.
```go
type FooRes interface{ fooRes() }
...
type FooStream struct{ Data io.ReadCloser }
type FooJSON struct{ ... }

func (*FooStream) fooRes() {}
func (*FooJSON) fooRes() {}
```

The problem is that if `FooStream` response is unhandled, it causes a leak.
```go
resp, err := client.Foo(ctx)
if err != nil { ... }

data, ok := resp.(*api.FooJSON)
if !ok { 
   // resp is [api.FooStream] and it is unclosed.
   // linters would not handle this case.
   return
}
```

I think `ogen` should generate a no-op `Close` method for every operation that has stream response.
```go
type FooRes interface {
    fooRes()
    Close() error     
}
```
or for all responses, if this feature is enabled.

cc @ernado

Reply:
I think same, `Close` should be in interface in this case.